src/ScheduleFunctions.cpp

```diff
@@ -673,7 +673,19 @@ class InjectRealization : public IRMutator {
     Stmt build_pipeline(Stmt s) {
         pair<Stmt, Stmt> realization = build_production(func);
 
-        return ProducerConsumer::make(func.name(), realization.first, realization.second, s);
+        Stmt producer;
+        if (realization.first.defined() && realization.second.defined()) {
+            producer = ProducerConsumer::make(func.name(), true, Block::make(realization.first, realization.second));
+        } else if (realization.first.defined()) {
+            producer = ProducerConsumer::make(func.name(), true, realization.first);
+        } else {
+            internal_assert(realization.second.defined());
+            producer = ProducerConsumer::make(func.name(), true, realization.second);
+        }
```

Comment:
I think the `ProducerConsumer::make` call should be lifted out of these if/else statements, so there is just one call that passes a `Stmt` constructed from the pair of Stmts.